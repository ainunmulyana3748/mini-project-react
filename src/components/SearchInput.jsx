// components/SearchInput.jsx
import React from "react";
import { Search } from "lucide-react";

const SearchInput = ({
  value,
  onChange,
  placeholder = "Cari...",
  autoFocus = false,
}) => {
  return (
    <div className="relative w-full max-w-md">
      <span className="absolute inset-y-0 left-0 pl-3 flex items-center pointer-events-none text-gray-400">
        <Search size={18} />
      </span>
      <input
        type="text"
        value={value}
        onChange={onChange}
        placeholder={placeholder}
        autoFocus={autoFocus}
        className="w-full pl-10 pr-4 py-2 border border-gray-300 rounded-xl shadow-sm focus:outline-none focus:ring-2 focus:ring-blue-500 focus:border-transparent transition"
      />
    </div>
  );
};

export default SearchInput;
