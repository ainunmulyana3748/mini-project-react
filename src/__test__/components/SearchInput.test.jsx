import { render, screen, fireEvent } from "@testing-library/react";
import SearchInput from "../../components/SearchInput";

describe("SearchInput Component", () => {
  test("render input dengan placeholder", () => {
    render(
      <SearchInput value="" onChange={() => {}} placeholder="Cari sesuatu..." />
    );
    expect(screen.getByPlaceholderText("Cari sesuatu...")).toBeInTheDocument();
  });

  test("onChange dijalankan saat mengetik", () => {
    const handleChange = jest.fn();
    render(<SearchInput value="" onChange={handleChange} />);
    fireEvent.change(screen.getByPlaceholderText("Cari..."), {
      target: { value: "abc" },
    });
    expect(handleChange).toHaveBeenCalled();
  });

  test("snapshot search input", () => {
    const { container } = render(<SearchInput value="" onChange={() => {}} />);
    expect(container).toMatchSnapshot();
  });
});
