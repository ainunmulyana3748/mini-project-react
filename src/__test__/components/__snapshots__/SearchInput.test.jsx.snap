// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput Component snapshot search input 1`] = `
<div>
  <div
    class="relative w-full max-w-md"
  >
    <span
      class="absolute inset-y-0 left-0 pl-3 flex items-center pointer-events-none text-gray-400"
    >
      <svg
        aria-hidden="true"
        class="lucide lucide-search"
        fill="none"
        height="18"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="18"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m21 21-4.34-4.34"
        />
        <circle
          cx="11"
          cy="11"
          r="8"
        />
      </svg>
    </span>
    <input
      class="w-full pl-10 pr-4 py-2 border border-gray-300 rounded-xl shadow-sm focus:outline-none focus:ring-2 focus:ring-blue-500 focus:border-transparent transition"
      placeholder="Cari..."
      type="text"
      value=""
    />
  </div>
</div>
`;
